docs/website/content/snippets/elasticnet.md

```diff
@@ -0,0 +1,19 @@
++++
+title = "Elastic Net"
++++
+```rust
+let (train, valid) = linfa_datasets::diabetes()
+    .split_with_ratio(0.9);
+
+// train pure LASSO model with 0.1 penalty
+let model = ElasticNet::params()
+    .penalty(0.1)
+    .l1_ratio(1.0)
+    .fit(&train)?;
+
+println!("z score: {:?}", model.z_score());
+
+// validate
+let y_est = model.predict(&valid);
+println!("predicted variance: {}", valid.r2(&y_est));
```

Comment:
Should this be `y_est.r2(&valid)`? The current r2 implementation assumes that the function argument is the ground truth and that `self` is the predicted values array

Reply:
the `r2` score is symmetric, but perhaps we are running into implementation issues, because `valid` uses multiple targets? Should definitely be `y_est.r2(&valid)?` though because this may fail

Reply:
You are right about the r2, sorry 😅 
It should have the `?` now because with the recent changes now it returns a `Result`, but the only difference between `y_est.r2(&valid)?` and `valid.r2(&y_est)?` I believe would be that one is `r2` and the other is `[r2]`